backend/app/services/attachment/parser.py

```
@@ -0,0 +1,383 @@
+            if not self.validate_text_length(text):
+                raise DocumentParseError(
+                    f"Extracted text exceeds maximum length ({self.MAX_TEXT_LENGTH} characters)"
+                )
+
+            return text, len(text), image_base64
+            
+        except DocumentParseError:
+            raise
+        except Exception as e:
+            logger.error(f"Error parsing document: {e}", exc_info=True)
+            raise DocumentParseError(f"Failed to parse document: {str(e)}")
```

Ruff (0.14.7)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)